tests/performance-tests/src/auth/getAuthDetails.ts

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env node
+
+/* eslint-disable camelcase */
+import puppeteer, { Page } from 'puppeteer';
+
+export interface AuthTokens {
+  accessToken: string;
+  refreshToken: string;
+  expiryDate: Date;
+}
+
+export interface AuthDetails {
+  userName: string;
+  adminUrl: string;
+  supportsRefreshTokens: boolean;
+  authTokens: AuthTokens;
+}
+
+export type IdpOption = 'azure' | 'keycloak';
+
+export const getAuthTokens = async (
+  email: string,
+  password: string,
+  adminUrl: string,
+  idp: IdpOption,
+): Promise<{
+  id_token: string;
+  access_token: string;
+  refresh_token: string;
+  expires_at: Date;
+}> => {
+  const browser = await puppeteer.launch({
+    headless: true,
+    ignoreHTTPSErrors: true,
+    product: 'chrome',
+    executablePath: '/usr/bin/google-chrome',
+    args: ['--no-sandbox', '--disable-setuid-sandbox'],
+  });
+
+  const page = await browser.newPage();
+
+  await page.goto(adminUrl);
+
+  await page.waitForXPath('//*[.="Sign in"]', {
+    timeout: 5000,
+  });
+
+  await page.click('button[class="govuk-button govuk-button--start"]');
+
+  try {
+    switch (idp) {
+      case 'azure':
+        await getAuthTokensAzure(page, email, password);
+        break;
+      case 'keycloak':
+        await getAuthTokensKeycloak(page, email, password);
+        break;
+      default:
+        throw new Error(`No login strategy for IDP ${idp}`);
+    }
+
+    return await page.evaluate(() => {
+      /* eslint-disable-next-line no-plusplus */
+      for (let i = 0; i < localStorage.length; i++) {
```

Comment:
This should get rid of the lint error: `/* eslint-disable-next-line no-plusplus */`

```suggestion
      for (let i = 0; i < localStorage.length; i+= 1) {
```

Reply:
Thanks, done